src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderSettingsDaoTest.java

```diff
@@ -29,46 +32,106 @@ class SnapshotBuilderSettingsDaoTest {
   @Autowired private DaoOperations daoOperations;
   @Autowired private SnapshotBuilderSettingsDao snapshotBuilderSettingsDao;
   private Dataset dataset;
+  private Snapshot snapshot;
 
   @BeforeEach
   void setup() throws IOException {
-    dataset = daoOperations.createMinimalDataset();
-    snapshotBuilderSettingsDao.upsertSnapshotBuilderSettingsByDataset(
-        dataset.getId(), SnapshotBuilderTestData.SETTINGS);
+    dataset = daoOperations.createDataset("dataset-minimal.json");
+    snapshotBuilderSettingsDao.upsertByDatasetId(dataset.getId(), SnapshotBuilderTestData.SETTINGS);
+    snapshot = daoOperations.createAndIngestSnapshot(dataset, "snapshot-from-dataset-minimal.json");
+    snapshotBuilderSettingsDao.upsertBySnapshotId(
+        snapshot.getId(), SnapshotBuilderTestData.SETTINGS);
```

Comment:
Knowing how long it takes for TDR to create a dataset, ingest data, and create a snapshot, it makes me wonder if there's a way to avoid running this every time we run any test in this class.

Some cases below don't require any of this setup and that seems especially wasteful, such as `getByDatasetIdThatDoesNotExistErrors()` and `getBySnapshotIdThatDoesNotExistErrors()`.